### PR TITLE
feat(node): added wp cli command - process pending requests

### PR DESCRIPTION
**Pull request by jaredrethman**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR introduces a WP CLI command for processing `np_webhook_request`s with post-meta status in `pending` state. 

### How to test the changes in this Pull Request:

1. Ensure a Hub and Node relationship exists and is configured correctly.
2. Node: Add incorrect "Hub URL" - WP Admin > Newspack Network > Node Settings
3. Node: Perform individual (as many as you can, the more the better) user related updates i.e. update; name, lastname etc.
4. Node: Add correct "Hub URL" - WP Admin > Newspack Network > Node Settings
5. Open terminal and run `wp newspack-network process-webhooks` to process and delete pending requests

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/node/class-webhook.php [hunk 1]
```diff
@@ -104,4 +110,98 @@ public static function sign( $data, $nonce, $secret_key = null ) {
...
+			++$processed_count;
+			$progress->tick();
+		}
+
+		$progress->finish();
+
+		if ( ! empty( $unprocessed_request_ids ) ) {
+			WP_CLI::warning( "The following requests could not be deleted:\n" . wp_json_encode( $unprocessed_request_ids ) );
+		}
+		WP_CLI::success( "Successfully processed {$processed_count}/{$requests_count} pending requests." );
```
Comment (chickenn00dle) [inline includes/node/class-webhook.php]: We probably want to do an additional check to confirm we've actually processed any requests before outputting the success message. I'd say if processed_count is 0 we should even output an error message. 

For example if I run this without fixing my hub URL the output looks like this:
Screenshot 2024-02-22 at 16 04 11

File: includes/node/class-webhook.php [hunk 2]
```diff
@@ -104,4 +110,98 @@ public static function sign( $data, $nonce, $secret_key = null ) {
...
+				}
+			}
+			++$processed_count;
+			$progress->tick();
+		}
+
+		$progress->finish();
+
+		if ( ! empty( $unprocessed_request_ids ) ) {
+			WP_CLI::warning( "The following requests could not be deleted:\n" . wp_json_encode( $unprocessed_request_ids ) );
```
Comment (chickenn00dle) [inline includes/node/class-webhook.php]: I think this message is meant to say '...could not be processed' vs. '...could not be deleted' considering we already delete warnings above.

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Works great @jaredrethman! Nice work 🎉 

I left two small comments regarding output, then I think this is good to go.

File: includes/node/class-webhook.php (hunk 1 above)
Comment (jaredrethman) [reply includes/node/class-webhook.php]: Have converted `warning` to `error` and added return after. Great suggestion!

File: includes/node/class-webhook.php (hunk 2 above)
Comment (jaredrethman) [reply includes/node/class-webhook.php]: Right you are, fixed, nice catch! 🤦

File: includes/node/class-webhook.php [hunk 3]
```diff
@@ -104,4 +110,99 @@ public static function sign( $data, $nonce, $secret_key = null ) {
...
+			}
+			++$processed_count;
+			$progress->tick();
+		}
+
+		$progress->finish();
+
+		if ( ! empty( $unprocessed_request_ids ) ) {
+			WP_CLI::error( "The following requests could not be processed:\n" . wp_json_encode( $unprocessed_request_ids ) );
+			return;
```
Comment (chickenn00dle) [inline includes/node/class-webhook.php:222]: Nit. Feel free to ignore. I think we were okay with the warning here. For large batches of events where some fail and some succeed, I would expect to see information about what failed and what succeeded.

Thinking about this some more, something like the following would be ideal IMO:

success - all events processed
warning - some events processed, some events not processed
error - no events processed

**Review by chickenn00dle — APPROVED** [review]
Thanks for the quick turnaround! I have one small nit, but since the current messaging works and I don't wanna hold this up, gonna go ahead and approve.

File: includes/node/class-webhook.php (hunk 3 above)
Comment (jaredrethman) [reply includes/node/class-webhook.php:222]: Some solid suggestions there @chickenn00dle! I have implemented your suggestions, capture last error (i.e. the one that caused the error to be captured) and added some additional comments/docs etc 36049d408ed5d461f43049cbf192678a2d2b8ba3, all-in-all your suggestion adds a lot of value to the command, thank you 🤜 🤛  

## Success:
Screenshot 2024-02-22 at 17 37 05

## Warning: (hard to test)
Screenshot 2024-02-22 at 17 39 10

## Errors:
Screenshot 2024-02-22 at 17 40 29

File: includes/node/class-webhook.php [hunk 4]
```diff
@@ -38,6 +41,9 @@ class Webhook {
 	public static function init() {
 		add_action( 'init', [ __CLASS__, 'register_endpoint' ] );
 		add_filter( 'newspack_webhooks_request_body', [ __CLASS__, 'filter_webhook_body' ], 10, 2 );
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			WP_CLI::add_command( 'np-network process', [ __CLASS__, 'cli_process' ] );
```
Comment (adekbadek) [inline includes/node/class-webhook.php]: We use the full plugin slug as the WP CLI command base elsewhere in code. Also, the command name should be more descriptive:

 ```suggestion
				WP_CLI::add_command( 'newspack-network process-webhooks', [ __CLASS__, 'cli_process' ] );
```

**Review by adekbadek — CHANGES_REQUESTED** [review]

File: includes/node/class-webhook.php (hunk 4 above)
Comment (jaredrethman) [reply includes/node/class-webhook.php]: Updated with your suggestions.

**Review by adekbadek — APPROVED** [review]

**Pull request merged**